homeassistant/components/recorder/migration.py

```diff
@@ -41,20 +41,39 @@ def _apply_update(engine, new_version):
     from sqlalchemy import Table
     from . import models
 
-    if new_version == 1:
-        def create_index(table_name, column_name):
-            """Create an index for the specified table and column."""
-            table = Table(table_name, models.Base.metadata)
-            name = "_".join(("ix", table_name, column_name))
-            # Look up the index object that was created from the models
-            index = next(idx for idx in table.indexes if idx.name == name)
-            _LOGGER.debug("Creating index for table %s column %s",
-                          table_name, column_name)
-            index.create(engine)
-            _LOGGER.debug("Index creation done for table %s column %s",
-                          table_name, column_name)
+    def create_index(table_name, column_name):
+        """Create an index for the specified table and column."""
+        table = Table(table_name, models.Base.metadata)
+        name = "_".join(("ix", table_name, column_name))
```

Comment:
The `index = next(...)` bits are a bit new to me. Does that effectively mean that any arguments passed from the function end up getting passed to an Index constructor? Asking because if we can _only_ specify columns, we may still have an issue if we wanted to, for instance, create a unique index where we would have to pass "Unique=true". If it's just blindly passing all the arguments though, that shouldn't be a problem? If it's only column names, that would be a problem (granted that I haven't solved in my code updates to this point yet either way).